packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -485,6 +374,89 @@ const DropdownMenuArrow = React.forwardRef<DropdownMenuArrowElement, DropdownMen
 
 DropdownMenuArrow.displayName = ARROW_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * DropdownMenuSub
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DropdownMenuSubProps {
+  children?: React.ReactNode;
+  open?: boolean;
+  defaultOpen?: boolean;
+  onOpenChange?(open: boolean): void;
+}
+
+const DropdownMenuSub: React.FC<DropdownMenuSubProps> = (
+  props: ScopedProps<DropdownMenuSubProps>
+) => {
+  const { __scopeDropdownMenu, children, open: openProp, onOpenChange, defaultOpen } = props;
+  const menuScope = useMenuScope(__scopeDropdownMenu);
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  return (
+    <MenuPrimitive.Sub {...menuScope} open={open} onOpenChange={setOpen}>
+      {children}
+    </MenuPrimitive.Sub>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * DropdownMenuSubTrigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_TRIGGER_NAME = 'DropdownMenuSubTrigger';
+
+type DropdownMenuSubTriggerElement = React.ElementRef<typeof MenuPrimitive.SubTrigger>;
+type MenuSubTriggerProps = Radix.ComponentPropsWithoutRef<typeof MenuPrimitive.SubTrigger>;
+interface DropdownMenuSubTriggerProps extends MenuSubTriggerProps {}
+
+const DropdownMenuSubTrigger = React.forwardRef<
+  DropdownMenuSubTriggerElement,
+  DropdownMenuSubTriggerProps
+>((props: ScopedProps<DropdownMenuSubTriggerProps>, forwardedRef) => {
+  const { __scopeDropdownMenu, ...subTriggerProps } = props;
+  const menuScope = useMenuScope(__scopeDropdownMenu);
+  return <MenuPrimitive.SubTrigger {...menuScope} {...subTriggerProps} ref={forwardedRef} />;
+});
+
+DropdownMenuSubTrigger.displayName = SUB_TRIGGER_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * DropdownMenuSubContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_CONTENT_NAME = 'DropdownMenuSubContent';
+
+type DropdownMenuSubContentElement = React.ElementRef<typeof MenuPrimitive.Content>;
+interface DropdownMenuSubContentProps extends MenuContentProps {}
```

Comment:
Now that we're exposing this we can do some work in `Menu` to simplify the nesting logic, this will also allow us to provide accurate prop types here. I'll do this separately after both PRs are merged.

Reply:
You mean adding `MenuSubContent` there as well right?

Reply:
Yes, It [already has the part](https://github.com/radix-ui/primitives/blob/6c398d9810fc52577780b5d987d2a6f11e86276b/packages/react/menu/src/Menu.tsx#L250), we should expose it and remove the nesting there too.